src/Akka.Cluster.Hosting.Tests/ClusterShardingDistributedDataSpecs.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Threading.Tasks;
+using Akka.Actor;
+using Akka.DistributedData;
+using Akka.Hosting;
+using Akka.Remote.Hosting;
+using FluentAssertions;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Akka.Cluster.Hosting.Tests;
+
+public class ClusterShardingDistributedDataSpecs: Akka.Hosting.TestKit.TestKit
+{
+    private const string ReplicatorName = "dDataReplicator";
+    
+    public ClusterShardingDistributedDataSpecs(ITestOutputHelper output): base(output: output)
+    {
+    }
+    
+    protected override void ConfigureAkka(AkkaConfigurationBuilder builder, IServiceProvider provider)
+    {
+        builder
+            .WithRemoting()
+            .WithClustering()
+            .WithDistributedData(opt =>
+            {
+                opt.Name = ReplicatorName;
+            });
+    }
+
+    [Fact(DisplayName = "WithDistributedData should start DistributedData extension automatically")]
+    public async Task WithDistributedDataStartsAutomaticallyTest()
+    {
+        var cluster = Cluster.Get(Sys);
+        cluster.Join(cluster.SelfAddress);
+        
+        var settings = ReplicatorSettings.Create(Sys);
+        var coordinatorName = settings.RestartReplicatorOnFailure ? $"{ReplicatorName}Supervisor" : ReplicatorName;
+        
+        var actorSelection = Sys.ActorSelection(new RootActorPath(cluster.SelfAddress) / "user" / coordinatorName);
+        
+        await AwaitAssertAsync(() =>
+        {
+            actorSelection.Tell(new Identify("coordinator"), TestActor);
+            var identity = ExpectMsg<ActorIdentity>(TimeSpan.FromSeconds(3));
```

Comment:
should really be using the `Task`-returning variants of these methods

Reply:
Done